File: src/main/java/ru/pgk/pgk/common/dto/PageDto.java
package ru.pgk.pgk.common.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.data.domain.Page;

import java.util.List;

@Getter
@AllArgsConstructor
public class PageDto<T> {
    private List<T> content;
    private Integer totalPage;
    private Boolean last;
    private Boolean first;

    public static <T>PageDto<T> fromPage(Page<T> page) {
        return new PageDto<>(page.getContent(), page.getTotalPages(), page.isLast(), page.isFirst());
    }
}

File: teacher-service/src/main/kotlin/ru/pgk/teacher_service/common/dto/PageDto.kt
package ru.pgk.teacher_service.common.dto

import org.springframework.data.domain.Page

data class PageDto<T>(
    val content: List<T>,
    val totalPage: Int,
    val last: Boolean ,
    val first: Boolean
)

fun <T>Page<T>.toDto(): PageDto<T> {
    return PageDto(this.content, this.totalPages, this.isLast, this.isFirst)
}